docs/source/serving/autoscaling.rst

```diff
@@ -119,3 +119,7 @@ If your service might experience long periods of time with no traffic, consider
     # ...
 
 The service will scale down all replicas when there is no traffic to the system and will save costs on idle replicas. When upscaling from zero, the upscale delay will be ignored in order to bring up the service faster.
+
+.. tip::
+
+    If the scale-to-zero is set, the clients that access the endpoint should make sure to have a retry mechanism to be able to wait until the replicas are provisioned and ready.
```

Comment:
```suggestion
    If the scale-to-zero is set, the clients that access the endpoint should make sure to have a retry mechanism to be able to wait until the replicas are provisioned and ready, i.e., starting a new replica when there is zero replica available.
```

Question: how does our autoscaler handle the case for retrying? If the client keeps retrying for failed requests, will those failed requests be considered for calculating the `target_num_replicas`? Need to think about the case, when zero replica is available and the client keeps retrying, causing the number of requests in the window become significantly large. Will it cause the autoscaler to scale up the service to `max_replicas`?

Reply:
Yes. The failed requests will contribute to calculating the `target_num_replicas`. 
It might cause the autoscaler to scale up to `max_replicas`. As long as the retry frequency is not too big we should be fine

Reply:
Should we avoid the request to make `target_num_replicas` to be larger than 1 when there is no ready replicas?

Reply:
I still think maybe we should not - SkyServe as it is now cannot differentiate between retry requests and actual user requests. When there is no ready replica but accumulated requests, it is possible that there is a big increase in user requests and we might want to scale to more replicas (even though some of them might be retries) to be safe.